docs/src/pages/docs/agents/01-agent-memory.mdx

```diff
@@ -7,6 +7,91 @@ description: Documentation on how agents in Mastra use memory to store conversat
 
 Agents in Mastra have a sophisticated memory system that stores conversation history and contextual information. This memory system supports both traditional message storage and vector-based semantic search, enabling agents to maintain state across interactions and retrieve relevant historical context.
 
+## How Memory Works
+
+The Memory API uses two main mechanisms to maintain context in conversations:
+
+### Recent Message History
+
+The `lastMessages` setting maintains a sliding window of the most recent messages. This ensures your agent always has access to the immediate conversation context:
+
+```tsx
+const memory = new Memory({
+  options: {
+    lastMessages: 10, // Keep 10 most recent messages
+  },
+});
+
+// Example conversation flow:
+// 1. User asks about project timeline
+await agent.stream('When will the project be completed?', {
+  threadId: 'project_123',
+  resourceId: 'user_123',
+  memoryOptions: {
+    lastMessages: 10,
+  },
+});
+
+// 2. User asks about specific feature
+await agent.stream('How will the new search feature work?', {
+  threadId: 'project_123',
+  resourceId: 'user_123',
+  memoryOptions: {
+    lastMessages: 10,
+  },
+});
+
+// 3. When user asks this question, the agent will see:
+//    - The last 10 messages from the conversation
+//    - Any relevant messages about search features from earlier
+await agent.stream('Can you summarize the search feature requirements?', {
+  threadId: 'project_123',
+  resourceId: 'user_123',
+  memoryOptions: {
+    lastMessages: 10,
```

Comment:
doesn't seem to include the APis?

Reply:
This section is just about `lastMessages` but I'll look at making that clearer - other APIs are covered further down

Reply:
Tweaked this in be05f07d2fece4b60f7bd52df9fe818d89c92ca0